src/model/pwrmgmt.go

```diff
@@ -23,15 +25,79 @@ var scalProfilesMap = map[string]ScalProfiles{
 type CpuGovernanceRule struct {
 	CPUs    string `yaml:"cpus"`
 	ScalGov string `yaml:"scaling_governor"`
+	MinFreq string `yaml:"min_freq"`
+	MaxFreq string `yaml:"max_freq"`
 }
 
 func (c CpuGovernanceRule) Validate() error {
-	if _, ok := scalProfilesMap[c.ScalGov]; !ok {
+	if _, err := cpulists.Parse(c.CPUs); err != nil {
+		return err
+	}
+	if _, ok := scalProfilesMap[c.ScalGov]; !ok && c.ScalGov != "" {
 		return fmt.Errorf("invalid cpu scaling governor: %v", c.ScalGov)
 	}
-	_, err := cpulists.Parse(c.CPUs)
-	if err != nil {
-		return fmt.Errorf("invalid cpus: %v", err)
+	min, errMin := ParseFreq(c.MinFreq)
+	if errMin != nil {
+		return fmt.Errorf("invalid min frequency: %v", errMin)
+	}
+	max, errMax := ParseFreq(c.MaxFreq)
+	if errMax != nil {
+		return fmt.Errorf("invalid max frequency: %v", errMax)
+	}
+
+	if min == -1 && max == -1 { // No frequency limits set, nothing to validate
+		return nil
```

Comment:
This shouldn't be done in a function responsible for other things. This technical dept will need to be reworked once another field gets added or needs validation afterwards.

Please move into its own function or scope the remaining check, without returning.